src/pudl/analysis/ml_tools.py

```diff
@@ -0,0 +1,181 @@
+from pydantic import BaseModel
+
+import pudl
+from pudl.workspace.setup import PudlPaths
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def get_model_config(experiment_name: str) -> dict:
+    """Load model configuration from yaml file."""
+    config_file = (
+        importlib.resources.files("pudl.package_data.settings")
+        / "record_linkage_model_config.yml"
+    )
+    config = yaml.safe_load(config_file.open("r"))
+
+    if not (model_config := config.get(experiment_name)):
+        raise RuntimeError(f"No {experiment_name} entry in {config_file}")
+
+    return {experiment_name: model_config}
+
+
+def _flatten_model_config(model_config: dict) -> dict:
+    """Take nested dictionary defining model config and flatten for logging purposes.
+
+    This is essentially a translation layer between Dagster configuration and mlflow,
+    which does not support displaying nested parameters in the UI.
+
+    Examples:
+        >>> _flatten_model_config(
+        ...     {
+        ...         'ferc_to_ferc': {
+        ...             'link_ids_cross_year': {
+        ...                 'compute_distance_matrix': {
+        ...                     'distance_threshold': .5,
+        ...                      'metric': 'euclidean',
+        ...                 },
+        ...                 'match_orphaned_records': {'distance_threshold': 0.5},
+        ...             }
+        ...         }
+        ...     }
+        ... ) == {
+        ...     'ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.distance_threshold': 0.5,
+        ...     'ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.metric': 'euclidean',
+        ...     'ferc_to_ferc.link_ids_cross_year.match_orphaned_records.distance_threshold': 0.5
+        ... }
+        True
+    """
+
+    def _flatten_level(config_level: dict, param_name: str):
+        flattened_dict = {}
+        for key, val in config_level.items():
+            flattened_param = f"{param_name}.{key}"
+            if isinstance(val, dict):
+                flattened_dict |= _flatten_level(val, param_name=flattened_param)
+            else:
+                flattened_dict[flattened_param[1:]] = val
+        return flattened_dict
+
+    return _flatten_level(model_config, "")
+
+
+class ExperimentTrackerConfig(Config):
+    """Dagster config to setup experiment tracking with mlflow."""
+
+    tracking_uri: str = f"sqlite:///{PudlPaths().output_dir}/experiments.sqlite"
+    tracking_enabled: bool = True
+    experiment_name: str
+    log_yaml: bool
+    run_context: str
+    #: Location to store artifacts. Artifact storage not currently used.
+    artifact_location: str = str(PudlPaths().output_dir)
+
+
+class ExperimentTracker(BaseModel):
+    """Class to manage tracking a machine learning model using MLflow.
+
+    The following command will launch the mlflow UI to view model results:
+    `mlflow ui --backend-store-uri {tracking_uri}`. From here, you can compare metrics
```

Comment:
It's kind of hard to figure out what the tracking_uri is, and then you have to actually go to this docstring to get the command for the MLflow UI. Could you spit out a logging message with the tracking URI and the command to open the UI? Maybe in `create_experiment_tracker`?

Reply:
Added a log